static/absolute/js/searchPage.js

```diff
@@ -1,8 +1,74 @@
 window.onload = function () {
   initMap();
   showRestaurants();
+  document.getElementById("search-btn").addEventListener("click", async () => {
+    const nameInput = document.getElementById("name-input");
+    const name = nameInput.value;
+
+    const addressInput = document.getElementById("address-input");
+    const address = addressInput.value;
+
+    let lat = null;
+    let long = null;
+
+    if (address === "") {
+      // Use Geolocation
+      try {
+        const {
+          coords: { latitude, longitude },
+        } = await getPosition({ enableHighAccuracy: true });
+        lat = latitude;
+        long = longitude;
+      } catch (err) {
+        alert("Failed to get position, please enter address.");
+        return;
+      }
+    } else {
+      // TODO(ved): Use API to convert address --> long/lat
+      alert("This is not yet supported");
+      return;
+    }
+
+    let data;
+    try {
+      const resp = await fetch(`api/${getEventId()}`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name,
+          location: [lat, long],
+        }),
+      });
+      data = await resp.json();
+    } catch (err) {
+      console.log(err);
+      alert("error posting to api");
+      return;
+    }
+
+    if (data.status !== 200) {
+      // TODO(ved): How should we display errors?
+      alert("error posting to api");
```

Comment:
I am not sure if this is a super helpful alert to display to the user. Maybe something a bit more more descriptive, like asking the user to try submitting again?

Reply:
you will definitely want better error handling in the future - quick and dirty is probably good enough for now to get things up and running though

Reply:
I can change the string -- I was assuming in the future we will display an error in the UI (and not give an alert). So this was more for debugging purposes.